cursor_pr_review.py

```
@@ -233,7 +233,138 @@ def _validate_anthropic(self) -> Dict[str, Any]:
+    def get_pr_details(self, repo: str, pr_number: str) -> Dict[str, Any]:
+        """Get PR details from GitHub API."""
+        try:
+            response = self.session.get(
+                f"https://api.github.com/repos/{repo}/pulls/{pr_number}",
+                headers={"Authorization": f"token {self.config.github_token}"}
+            )
+            response.raise_for_status()
+            return response.json()
+
+        except requests.exceptions.RequestException as e:
+            raise APIError(f"Failed to get PR details: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)